### DISCUSSION TITLE
Repeater and relationship data in form

### DISCUSSION THREAD

**Discussion by maverik23** [filamentphp/filament#14062] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I have a Resource (ticket) that has some fields and a repeater where the user can add items. These items are a relationship between the ticket and ticket_items.

When I add mutateFormDataBeforeSave() in the editTicket, I don't have the repeater data. I tried using mutateRelationshipDataBeforeSaveUsing(), but I still don't have the data.

My problem is that I need the items' data to calculate totals and other values. Does anyone know how I can get this data in the form's $data?

`Repeater::make('items')
                            ->label(' ')
                            ->schema([
                                TextInput::make('item.nombre')
                                    ->formatStateUsing(fn($record) => $record ? $record->nombre : '')
                                    ->required()
                                    ->maxLength(50),
                                TextInput::make('item.importe')
                                    ->formatStateUsing(fn($record) => $record ? $record->importe : '')
                                    ->required()
                                    ->numeric()
                                    ->maxValue(50000000),
                            ])->columns(4)->defaultItems(1)
                            ->relationship('ticketItems')`
                            
`protected function mutateFormDataBeforeSave(array $data): array
    {
        dd($data);
        return $data;
    }`
    
    
    What I need is to have the items' data in the form's array like this:
    `$data = [
    'name' => 'Jon Doe',
    'age' => '36',
    'items' => [
        'nombre' => 'item1',
        'importe' => '28.50',
    ],
];`

**Comment by leandrocfe** [comment]
If you want to save it manually, you can add `->dehydrated()` in the repeater after `->relationship()`

**Reply by maverik23** [reply]
thanks, it works fine.
